ibis/udf/validate.py

```diff
@@ -49,20 +92,23 @@ def validate_input_type(
     inspect.Signature
     """
     funcsig = signature(func)
-    declared_parameter_count = len(input_type)
-    function_parameter_count = _parameter_count(funcsig)
-
-    if declared_parameter_count != function_parameter_count:
-        raise TypeError(
-            'Function signature {!r} has {:d} parameters, '
-            'input_type has {:d}. These must match. Non-column '
-            'parameters must be defined as keyword only, i.e., '
-            'def foo(col, *, function_param).'.format(
-                func.__name__,
-                function_parameter_count,
-                declared_parameter_count,
+
+    if Parameter.VAR_POSITIONAL not in _parameter_kinds(funcsig):
+        declared_parameter_count = len(input_type)
+        function_parameter_count = _parameter_count(funcsig)
+
```

Comment:
do you have a test that exercises this one?

Reply:
[`test_invalid_kwargs`](https://github.com/ibis-project/ibis/blob/8b79ae53d424806a9aa270bd8c673777e492ba1d/ibis/tests/all/test_vectorized_udf.py#L132) tests that an exception is raised (because `declared_parameter_count  != function_parameter_count`)

Other tests in the module kind of indirectly test the two cases when the exception is not raised:

- [`test_elementwise_udf`](https://github.com/ibis-project/ibis/blob/8b79ae53d424806a9aa270bd8c673777e492ba1d/ibis/tests/all/test_vectorized_udf.py#L11): (no exception is raised because `declared_parameter_count  == function_parameter_count`)
- [`test_valid_args`](https://github.com/ibis-project/ibis/blob/8b79ae53d424806a9aa270bd8c673777e492ba1d/ibis/tests/all/test_vectorized_udf.py#L79): (has `*args`, so doesn't raise an exception even though `declared_parameter_count  != function_parameter_count` because the `if` statement is `False`)